Source/AssetRipper.Processing/AnimatorControllers/BidirectionalDictionary.cs

```diff
@@ -0,0 +1,138 @@
+namespace AssetRipper.Processing.AnimatorControllers
+{
+	/// <summary>
+	/// Bidirectional Dictionary, valid for 2 different Types T1 and T2
+	/// </summary>
+	public sealed class BidirectionalDictionary<T1, T2>
+		where T1:notnull
+		where T2:notnull 
+	{
+		private readonly Dictionary<T1, T2> forward = new();
+		private readonly Dictionary<T2, T1> backward = new();
+
+		public int Count { get { return forward.Count; } }
+
+		public bool IsReadOnly { get { return false; } }
+
+		public T2 this[T1 item1]
+		{
+			get
+			{
+				return forward[item1];
+			}
+			set
+			{
+				forward[item1] = value;
+				backward[value] = item1;
+			}
+		}
+
+		public T1 this[T2 item2]
+		{
+			get
+			{
+				return backward[item2];
+			}
+			set
+			{
+				forward[value] = item2;
+				backward[item2] = value;
+			}
+		}
+
+		public ICollection<T1> Keys { get { return forward.Keys; } }
+
+		public ICollection<T2> Values { get { return backward.Keys; } }
+
+		public void Add(T1 item1, T2 item2)
+		{
+			if (item1 == null)
+			{
+				throw new ArgumentNullException(nameof(item1));
+			}
+			if (item2 == null)
+			{
+				throw new ArgumentNullException(nameof(item2));
+			}
```

Comment:
I think there's a code fixer that can convert this to `ArgumentNullException.ThrowIf`.